src/eppo_value.ts

```diff
@@ -56,18 +90,22 @@ export class EppoValue {
   }
 
   static Bool(value: boolean): EppoValue {
-    return new EppoValue(ValueType.BoolType, value, undefined, undefined);
+    return new EppoValue(ValueType.BoolType, value, undefined, undefined, undefined);
   }
 
   static Numeric(value: number): EppoValue {
-    return new EppoValue(ValueType.NumericType, undefined, value, undefined);
+    return new EppoValue(ValueType.NumericType, undefined, value, undefined, undefined);
   }
 
   static String(value: string): EppoValue {
-    return new EppoValue(ValueType.StringType, undefined, undefined, value);
+    return new EppoValue(ValueType.StringType, undefined, undefined, value, undefined);
+  }
+
+  static JSON(value: string, typedValue: object): EppoValue {
+    return new EppoValue(ValueType.JSONType, undefined, undefined, value, typedValue);
   }
 
   static Null(): EppoValue {
-    return new EppoValue(ValueType.NullType, undefined, undefined, undefined);
+    return new EppoValue(ValueType.NullType, undefined, undefined, undefined, undefined);
```

Comment:
We could probably remove all the `?? null` operations in the get<Typed>Assignment methods if we used `null` instead of `undefined` for the empty arguments in the constructor. Go for it? Or is that too much refactoring?

Reply:
I like paranoia of `?? null` everywhere. Agree it would be more correct of us to pass in `null` for the other parameters. Although for something with a signature of five parameters, at that point it may make sense to just have an options object as the second parameter for `xxxValue`, and in that case we'd be getting `undefined` again.